### ISSUE TITLE
Can't set portal user access to 'contact' - no option

### ISSUE THREAD

**Issue by bandtank** [espocrm/espocrm#480]
The documentation says I should be able to limit access to a record by a contact relationship:

![image](https://cloud.githubusercontent.com/assets/1170415/24640807/507f52b6-18b7-11e7-9153-983bb736ed95.png)

However, I don't have that option:

![image](https://cloud.githubusercontent.com/assets/1170415/24640814/62646fa2-18b7-11e7-9910-d6ed1f7fe9a5.png)

What am I doing wrong?

**Comment by yurikuzn** [comment]
You can add contact in metadata. Example for Task: https://github.com/espocrm/espocrm/blob/master/application/Espo/Modules/Crm/Resources/metadata/scopes/Task.json#L7

**Comment by yurikuzn** [comment]
Is it a custom entity?

**Comment by bandtank** [comment]
Yes, it is a custom entity. What I really want to do is assign a custom entity as the scope for another custom entity. For example, I have an entity called Vendor which stores subcontractors, printing companies, delivery companies, or any other person or company who performs a service for my business. It doesn't make sense to store them as Clients because we are hiring them and clients hire us. My Project entity is intended to be visible through multiple portals to Clients (the company who hired us to do work) and Vendors (the companies we hire to help us do the project).

I setup two portals - one for Client access and another for Vendor access. It's simple to allow Clients and the associated portal users to access their own projects, but I can't figure out how to allow a Vendor to access the same project and also edit certain fields. I think what I ultimately need is a way to set the ACL scope to vendor, which would allow me to give access to a Vendor portal user based on a Project<->Vendor relationship just like Client portal users can access Projects based on a Project<->Vendor relationship. Is that even possible? I can't figure out how to accomplish this.

**Comment by yurikuzn** [comment]
Not sure how difficult but anyway it requires some customization. In Acl, Select Manager. It's not designed to support custom acl levels.

**Comment by yurikuzn** [comment]
You can use the same `contact` level access but in Acl and SelectManager implementations do a proper check.
